tests/system/projects/test_project.py

```diff
@@ -1067,6 +1068,52 @@ def _assert_workflow_status(workflow, status):
             mlrun.run.RunStatuses.failed,
         )
 
+    def test_project_default_function_node_selector(self):
+        project_name = "test-project"
+        function_name = "test-func"
+        project_label_name, project_label_val = "kubernetes.io/arch", "amd64"
+        function_label_name, function_label_val = "kubernetes.io/os", "linux"
+
+        igz_mgmt.Project.create(
+            self._igz_mgmt_client,
+            name=project_name,
+            owner="admin",
+            default_function_node_selector=[
+                {"name": project_label_name, "value": project_label_val}
+            ],
+        )
+
+        project = self._run_db.get_project(project_name)
+        assert project.spec.default_function_node_selector == {
+            project_label_name: project_label_val
+        }
+
+        code_path = str(self.assets_path / "sleep.py")
+        func = project.set_function(
+            name=function_name,
+            func=code_path,
+            kind="job",
+            image="mlrun/mlrun",
+            handler="handler",
+        )
+        func.spec.node_selector = {function_label_name: function_label_val}
+
+        # We run the function to ensure node selector enrichment, which doesn't occur during function build,
+        # but at runtime.
+        project.run_function(function_name)
+
+        # Verify that the node selector is correctly enriched
+        result_func = project.get_function(function_name)
+        assert result_func.spec.node_selector == {
+            project_label_name: project_label_val,
+            function_label_name: function_label_val,
+        }
+
+        self._run_db.delete_project(
+            name=project_name,
+            deletion_strategy=mlrun.common.schemas.DeletionStrategy.cascade,
+        )
```

Comment:
use `self.custom_project_names_to_delete.append(name)` if you create a project within the test case itself, that way we can track and delete on teardown